Upgrade wazuh-agent to 4.14.1 version

kinseii opened the pull request:
Subj.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added Amazon Linux and Ubuntu agent image variants.
  * Streamlined agent registration, cleanup, and deregistration flows with configurable SSL handling.
  * Improved service startup ordering and health-check reliability with enhanced diagnostics.

* **Chores**
  * Upgraded Wazuh agent to 4.14.1 across images and compose setups.
  * Updated dependency pins and build/image defaults for reproducibility.
  * Hardened CI and image-build workflows for more stable, repeatable releases.

* **Tests**
  * Increased resilience of integration tests with retries and expanded test matrices.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kinseii committed:
Update Wazuh Agent version to 4.14.1-1

kinseii committed:
Fix workflows to use DOCKER_HUB_USERNAME secret

kinseii committed:
Fix Ubuntu release to 24.04 for pip compatibility

kinseii committed:
Fix pip install conflict on Ubuntu 24.04 with --ignore-installed

kinseii commented on .github/workflows/docker-hub-image-main.yml:
```diff
@@ -14,17 +14,11 @@ jobs:
-            image: opennix/wazuh-agent-amazonlinux
-          - dockerfile: ./images/Dockerfile.ubuntu
-            image: opennix/wazuh-agent-ubuntu
+            image: wazuh-agent-ubuntu
```
@pyToshka 
Should I return this logic?

kinseii committed:
Use a pinned Ubuntu image digest

kinseii committed:
Revert multiple reposititories logic

kinseii committed:
Remove version suffix

kinseii committed:
Fix SonarQube issue: Line is too long. Split it into multiple lines using backslash continuations

kinseii committed:
Add personal repository for checking

kinseii committed:
Fix SonarQube issue: Use full commit SHA hash

kinseii committed:
Fix issue: Deprecation notice - v3 of the artifact actions

kinseii committed:
Update versions Scorecard

kinseii committed:
Update packages versions of docker and git hubs tools

kinseii committed:
Update packages versions for snyk

kinseii committed:
Fix issue: CodeQL Action v3 will be deprecated in December 2026

kinseii committed:
Fix Snyk alert: hardcoded 'verify=False'

kinseii committed:
Fix Snyk alert: hardcoded 'verify=False' in yet 2 files. Fix SonarQube alert with list method

kinseii committed:
Fix SonarCube duplication alert: Refactoring req and auth logic

kinseii committed:
Fix SonarCube duplication alert: Refactoring del logic

kinseii committed:
Fix SonarCube alert: Replacing unused status_code variable with underscore. Remove duplication code. Refactoring for improve readable

kinseii committed:
Fix StoreCard alert: Update python libs. Refactoring tests permissions

kinseii committed:
Fix SonarQube alert: Applying SHA pins for tests packages

kinseii committed:
Fix StoreCard alert: Updating requests and urllib3 versions

kinseii committed:
Fix StoreCard alert: Updating python libs versions to latest

kinseii commented:
@pyToshka 

I'm not sure what the best approach is. By removing checks: write, we will most likely lose the nice annotations in the “Checks” / “Files Changed” tab for test results.

<img width="650" height="188" alt="image" src="https://github.com/user-attachments/assets/0ea57230-21e6-49e4-bbde-2eb503b6a151" />

I fixed the rest of the alerts.

pyToshka commented:
> @pyToshka
> 
> I'm not sure what the best approach is. By removing checks: write, we will most likely lose the nice annotations in the “Checks” / “Files Changed” tab for test results.
> 
> <img alt="image" width="650" height="188" src="https://private-user-images.githubusercontent.com/74458264/525208485-0ea57230-21e6-49e4-bbde-2eb503b6a151.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU0MzM3NzcsIm5iZiI6MTc2NTQzMzQ3NywicGF0aCI6Ii83NDQ1ODI2NC81MjUyMDg0ODUtMGVhNTcyMzAtMjFlNi00OWU0LWJiZGUtMmViNTAzYjZhMTUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjExVDA2MTExN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjNzNhYWMxZTg1OTFhYTcwYTBmY2FiMmI0ZDI0Y2U5NTY4OWU1NTg4YTIyYThjNGQ4MGM4NjU5YjU0OTA1ZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DPzx0pgzEjYXGXwvpm8pCgA9Ivpyr1Q_pKpzBPpYVaE">
> I fixed the rest of the alerts.

i think muting will fine

kinseii committed:
Refactor by CodeRabbit review

kinseii committed:
Permission fixes of workflow files

kinseii committed:
Refactor by CodeRabbit review 2

kinseii committed:
Removing print statements. Refactor by CodeRabbit review 3

kinseii committed:
Fix ScoreCard alert: Adding more hashes. Yet more more more

kinseii committed:
Add v4.11.1-1 and v4.14.1-1 to build tests

kinseii committed:
Separate server verions. Migrate to build-push-action

kinseii committed:
Separate server verions. Migrate to build-push-action

kinseii committed:
Update actions/setup-python to v6.1.0

kinseii committed:
Adding hash for black

kinseii committed:
Adding hash for black

kinseii committed:
Increase workers

kinseii committed:
Adding more hashes

kinseii committed:
Adding more more hashes

kinseii committed:
Adding more more more hashes

kinseii committed:
Downgrade iniconfig==2.1.0

kinseii committed:
Fix hash for exceptiongroup==1.3.0

kinseii committed:
Fix hash for few packages

kinseii committed:
Add hashes for both python 3.9 and 3.10

kinseii committed:
Switch tests to python v3.10

kinseii committed:
Add hash for six paackage

kinseii committed:
Regen hashes with pip-compile

kinseii committed:
Fix python version in the tests

kinseii committed:
Separate artifact names for tests

kinseii committed:
Pin actions/download-artifact

kinseii committed:
Fix tests: docker compose

kinseii committed:
Fix tests: Check Wazuh indexer start

kinseii committed:
Fix tests: Implementing retry logic

kinseii committed:
Fix tests alert: Increasing JVM heap

kinseii committed:
Fix tests alert: Adding diag logs

kinseii committed:
Fix tests alert: Python syntax fix

kinseii committed:
Move Print cotainer logs step ahead

kinseii committed:
Fix tests: Change wazuh versions

kinseii committed:
Fix tests: Change path for XML report

kinseii committed:
Fix tests: Remove useless cp XML report

kinseii committed:
Fix tests: Fix free space and cert name

kinseii committed:
Fix paths

kinseii committed:
Fix permissions

kinseii committed:
Fix permissions

kinseii committed:
Fix permissions

kinseii committed:
Fixes: health check and images saving

kinseii committed:
Fix CI: Add artifact handling and build fallback

- Add continue-on-error for artifact download (fallback to rebuild)
- Add diagnostic output to debug artifact issues
- Add --build flag back for fallback if image not found
- Keep timeout protection (10 min)

Fixes integration tests failing due to missing artifacts.

kinseii committed:
Increase timeout to 15min and parallelize image pulls

kinseii committed:
Increase timeout to 20 minutes for slow builds

kinseii committed:
Remove artifact logic, enable BuildKit for compose

kinseii committed:
Build image with buildx cache before compose up

kinseii committed:
Build all 3 agent images with cache before compose

kinseii committed:
Remove prune, increase timeout to 30min

kinseii committed:
Simplify healthcheck, increase retries and start period

kinseii committed:
fix: improve apt reliability with retries and fix-missing

- Use apt-get instead of apt (better for scripts)
- Add --fix-missing and Acquire::Retries=3 for mirror sync issues
- Add --no-install-recommends to reduce image size
- Replace deprecated apt-key with gpg --dearmor
- Fix FROM/AS case mismatch warning

kinseii committed:
fix: healthcheck now verifies manager API instead of indexer

- Check manager API on localhost:55000 instead of indexer cluster health
- Remove obsolete version attribute from docker-compose.yml

kinseii committed:
fix: healthcheck accepts any HTTP response from API

API uses JWT tokens, so basic auth on root fails with 401.
Now checks if API responds with any 2xx/3xx/4xx status code.

kinseii committed:
fix: mount indexer certs to /etc/wazuh-indexer/certs/

Image expects certs at /etc/wazuh-indexer/certs/ with names indexer.pem,
indexer-key.pem, root-ca.pem. Added mounts with correct paths.

kinseii committed:
fix: remove /etc/wazuh-indexer/certs mounts causing Java Security Manager issues

Mounts to /etc/wazuh-indexer/certs/ fail due to Java Security Manager policy.
Added config mount to /etc/wazuh-indexer/opensearch.yml as fallback.

kinseii committed:
Add healthcheck for wazuh.indexer and update depends_on conditions
- Add healthcheck that verifies cluster health is green or yellow
- Update depends_on for amazonlinux-agent, ubuntu-agent, and dashboard
- Use service_healthy condition to ensure proper startup order
- This should fix flaky tests where indexer not fully initialized

kinseii committed:
Remove service_healthy conditions from depends_on

The condition: service_healthy was causing docker-compose up -d to fail
quickly because it waits for healthchecks before starting dependent
services. This caused issues with older Wazuh cluster versions.

Healthchecks are kept for monitoring purposes but dependencies now use
simple startup order. The CI workflow already has a Check Wazuh indexer
start step that waits for readiness.

kinseii committed:
Accept yellow cluster status in indexer health check

Single-node OpenSearch clusters typically show yellow status because
there are no other nodes to allocate replica shards. This is normal
and the cluster is fully functional.

Updated the workflow to accept both green and yellow status, matching
the healthcheck behavior in docker-compose.yml.

kinseii committed:
Fix: prevent curl failures from exiting health check script

GitHub Actions runs bash with set -e by default, causing the script
to exit immediately when curl fails (connection refused during startup).

Added || true to curl commands to prevent early exit and allow the
retry loop to function properly.

kinseii committed:
Add security initialization for older Wazuh indexer versions

- Added admin certificate generation to certs.yml
- Added Initialize Wazuh indexer security step to workflow
- Runs securityadmin.sh to create .opendistro_security index
- This fixes integration tests for Wazuh 4.7.0 cluster version

kinseii committed:
Fix admin certificate IP in certs.yml

Use 127.0.0.1 instead of invalid "admin" IP for admin certificate generation.

kinseii committed:
Fix cert permissions

kinseii committed:
Add diagnostic output for indexer startup issues

- Show container status at start
- Print indexer logs every 5 attempts while waiting
- Print full logs if port never becomes available
- This will help identify why older Wazuh versions fail to start

kinseii committed:
Add detailed diagnostics for cluster health check

- Print health response on each attempt
- Show cluster info and security plugin status every 5 attempts
- Print full diagnostics on timeout including indexer logs

kinseii committed:
Trigger CI re-run

kinseii committed:
Add retry logic to Docker pull steps in CI workflow

Adds a retry_pull function that retries failed docker pulls up to 3 times
with 10 second delays between attempts. This helps handle transient
infrastructure failures from Docker Hub or GitHub Actions runners.

kinseii committed:
Add workflow_dispatch trigger for manual workflow runs

Allows manually triggering the Tests workflow from the GitHub Actions UI,
useful for re-running after transient infrastructure failures.

kinseii committed:
Fix docker commands to use docker compose for container access

Replace `docker exec` and `docker logs` with `docker compose exec` and
`docker compose logs` to correctly reference containers by service name
instead of container name (which includes project prefix).

kinseii committed:
Fix securityadmin.sh execution by setting JAVA_HOME

The securityadmin.sh script was failing silently because JAVA_HOME
was not set. Now we:
1. Detect Java location in the container
2. Set JAVA_HOME and PATH appropriately
3. Capture and display actual securityadmin.sh output
4. Check exit code before declaring success

kinseii committed:
Fix admin certificate placement in certs.yml

Move admin from indexer section to its own admin section within nodes.
This ensures the admin certificate is generated with the correct
parameters for use with securityadmin.sh.

kinseii committed:
Add certificate debugging and -icl flag

Debug certificate files to diagnose PKIX validation failure:
- List certificate files in mounted volume
- Show subject/issuer of root-ca, indexer, and admin certs
- Add -icl flag to ignore cluster name

kinseii committed:
Update cert generator to 0.0.4 (latest) and improve diagnostics

- Update wazuh-certs-generator from 0.0.1 to 0.0.4
- Use keytool instead of openssl for certificate diagnostics
- Show Owner/Issuer to verify certificate chain

kinseii committed:
Remove admin section from certs.yml

The admin certificate is auto-generated by wazuh-certs-generator.
Adding an explicit admin section causes 0.0.4 to fail.

kinseii committed:
Use wazuh-certs-generator 0.0.2

Version 0.0.4 has a bug - cant download wazuh-certs-tool.sh

kinseii committed:
Add diagnostic to check server certificate

Check what certificate the indexer is actually presenting to clients.

kinseii committed:
Add detailed OpenSSL chain verification diagnostics for PKIX debug

kinseii committed:
Check if security is already initialized before running securityadmin.sh

kinseii committed:
Add warning handling and retry for security initialization

kinseii committed:
Try JKS truststore for securityadmin.sh to fix PKIX on older versions

kinseii committed:
Use Wazuh indexer-security-init.sh and add --accept-red-cluster flag

kinseii committed:
Add extra wait time for older indexer versions (30s + increased health check iterations)

kinseii committed:
Add fail-fast: false to integration tests to see all results

kinseii committed:
Use 4.14.0 cluster for all agent tests (backward compatible)

Older clusters (4.7.0, 4.11.0) have security initialization issues that
prevent the indexer health check from passing. Agents are backward
compatible with newer clusters, so we use 4.14.0 for all tests.

kinseii commented on images/Dockerfile.ubuntu:
```diff
@@ -1,13 +1,13 @@
-ARG TAG=@sha256:145bacc9db29ff9c9c021284e5b7b22f1193fc38556c578250c926cf3c883a13
+ARG TAG=:24.04
```
omg, ok!
